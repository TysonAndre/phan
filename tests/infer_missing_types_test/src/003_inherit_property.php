<?php

namespace NS3;

use ArrayObject;

class Y extends X {

}
class Z extends Y {
    public function test() {
        // Phan should infer that $this->val is of type ArrayObject when --analyze-twice is used,
        // despite the assignment occurring later in analysis.
        $this->val->missingMethod();
    }
}
class X {
    protected $val;
    public function __construct(ArrayObject $val) {
        $this->val = $val;
    }
}
