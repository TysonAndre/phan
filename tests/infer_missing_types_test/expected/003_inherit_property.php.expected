src/003_inherit_property.php:10 PhanUnreferencedClass Possibly zero references to class \NS3\Z
src/003_inherit_property.php:11 PhanPluginUnknownMethodReturnType Method \NS3\Z::test() has no declared or inferred return type
src/003_inherit_property.php:11 PhanUnreferencedPublicMethod Possibly zero references to public method \NS3\Z::test()
src/003_inherit_property.php:14 PhanUndeclaredMethod Call to undeclared method \ArrayObject::missingMethod
src/003_inherit_property.php:18 PhanPluginUnknownPropertyType Property \NS3\X->val has an initial type that cannot be inferred (Types inferred after analysis: \ArrayObject)
